operations/sigmoidOperation_Impl.h

```diff
@@ -0,0 +1,49 @@
+/*
+    This file contains the implementation of the forward and backward pass of
+    the sigmoid operation.
+*/
+
+#include "operations/sigmoidOperation.h"
+
+#ifndef __OP_IMPL_SIG_INCLUDED__
+#define __OP_IMPL_SIG_INCLUDED__
+
+/* 
+    Backpropogation of the sigmoid operation. Example of a operation is as follows:
+    
+    F = sigmoid(x) is forward propogation
+    The gradients would be as follows:
+    1. dF/dx = (1 - F)*F
+
+    This formula can be derived by hand.
+*/
+template <typename T>
+void SigmoidOperation<T>::backward(Matrix<T> grad) {
+    auto g = ((T)1 + ((T)-1)*this->t3->val)*this->t3->val; // matrix can be overloaded
```

Comment:
Can we make this line of code simpler so that it is easier to understand?

Maybe by splitting it into multiple lines.

Reply:
I think it's pretty clear. And we're doing operator overloading to make expressions like these to fit in one line. I've added a comment that signifies what expression it is to make it clearer.